### ISSUE TITLE
Arbitrary Editors diff Comparison

### ISSUE THREAD

**Issue by yashpalgoyal1304** [lapce/lapce#2136]
## Is your feature request related to a problem? Please describe.

Need to see the difference i.e. compare two arbitrary buffers - i.e. in full generality.
- Buffers as it maybe unsaved new files too, not necessarily a saved file.
- Arbitrary as I am talking in context of non-SCM or non-R/VCS

## Describe the solution you'd like

Ability to select any two arbitrary editors in the "Open Editors" panel and "Compare selected".

This feature is available in VS Codium (and i [guess in Atom] too)

[guess in atom]: https://github.com/lapce/lapce/issues/687#issuecomment-1316264665

![VSCodium_NjmDopLxHB](https://user-images.githubusercontent.com/19423063/218265749-15bc33cd-4c90-4690-b813-703961f29613.png)

## Describe alternatives you've considered

The frontend is as simple as it can get, so, i don't think any alternates exist.

## Additional context

- https://github.com/lapce/lapce/issues/687#issuecomment-1426794223

**Comment by Jieiku** [comment]
Here is the link to the plugin that made this work for Atom: https://github.com/mupchrch/split-diff

**Comment by yashpalgoyal1304** [comment]
Being open source, and source of inspiration for Lapce interface (i guess), here's links to compare view or difference editors development in vs code:

* Diff editor review pane ([v1.14] 2017-06)
* Compare dirty file with version on disk ([v1.15] 2017-07)
* Compare open file to Clipboard ([v1.19] 2017-11)
* Multi select in the Explorer ([v1.20] 2018-01)


[v1.14]: https://code.visualstudio.com/updates/v1_14#_diff-editor-review-pane
[v1.15]: https://code.visualstudio.com/updates/v1_15?source=post_page---------------------------#_compare-dirty-file-with-version-on-disk
[v1.19]: https://code.visualstudio.com/updates/v1_19#_compare-open-file-to-clipboard
[v1.20]: https://code.visualstudio.com/updates/v1_20#_multi-select-in-the-explorer

+ Difference editor
https://code.visualstudio.com/docs/editor/command-line#_core-cli-options

**Comment by Jieiku** [comment]
I was marked as off topic, the requested feature here, is a plugin in Atom and that plugin was open source with MIT license (so possibly useful?)... I'm not sure how I was off topic... I get the feeling I'm not wanted here, so I will simply quit trying to help in this repository. I still hope Lapce is a blazing success but I won't bother wasting my time with bug reports or feature requests anymore. Going to find any threads Ive posted in this repository and unsubscribe because I can't bother taking the time to follow up on things if that's how its going to be handled.

**Comment by Jieiku** [comment]
@dzhou121 and @bugadani you might want to review the way that @panekj is moderating people. good luck to you all and this repo.

**Comment by panekj** [comment]
> is a plugin in Atom and that plugin was open source with MIT license (so possibly useful?)... I'm not sure how I was off topic...

It's marked as off-topic because this is a Rust language codebase with completely different design, so I don't see how mentioning a plugin for another editor would help anyhow

**Comment by Jieiku** [comment]
If you dont see how comparing similar features between other programs can be helpful, then I am really not sure what else to say... but like I said, I have unsubscribed from all my threads here, you won't have to deal with me anymore.

**Comment by panekj** [comment]
> you might want to review the way that @panekj is moderating people.

I'm not moderating people, I'm (trying) to moderate a bug tracker. I hide stuff that I think is not helpful and detracts from actual discussion

**Comment by yashpalgoyal1304** [comment]
hi @Jieiku , i can understand ur feelings...

i would say that 
- don't take it personally
- he hides his own comments as offtopic too (see attached screenshot below)

a better way could just have been to raise an objection, something like: 
"hey, i think my comment is very relevant to this issue for source of inspiration. can u please unhide it"

Different people have different way of categorising things, and often times, conflcts arise... i'd say to take it easy. 

Lots of things are happening all the times in issue trackers lol. people single handedly close ur issues or pulls as "fixed" when they are not or "discarded" without waiting for input from other side, or blatantly ignore ur input and don't consider it, etc etc....

Better would be to stay lite and don't pile such meager things to heart.

<img src="https://user-images.githubusercontent.com/19423063/218292896-896098a3-8f5e-4c1a-97b0-f8b4e956c4c3.png" width=600px/>

**Comment by yashpalgoyal1304** [comment]
I also think that it's partially due to lack of features in issue trackers.... it's all just one linear comment mode here...

i always wanted different tabs or some way to club (group comments as related) in issues 
so, "comments", "summary/conclusions", "inspirations", "implementations" or custom ones

**Comment by Jieiku** [comment]
Yes your right, for whatever reason it felt like a slap in the face and I just got off work. I did not realize they were marking their own posts like that too.... I can honestly say in my years of using github I have not seen it done that way, but there plenty of people more active on github than I am.

I apologize if I upset anyone, not my intention, we can end it with this... don't want to derail this feature request because it is a good one! @panekj again I apologize if I came off as harsh, I could have responded more calmly to my post being marked off topic.

**Comment by yashpalgoyal1304** [comment]
Now if we can hide all these as offtopic 😆🤣🤣 , just kiddin'

![lol]![rofl]

[lol]: https://github.githubassets.com/images/icons/emoji/unicode/1f606.png
[rofl]: https://github.githubassets.com/images/icons/emoji/unicode/1f923.png

**Comment by panekj** [comment]
No worries and I'm sorry too, it's just my habit from work :)

> I can honestly say in my years of using github I have not seen it done that way

I know few projects but it's definitely very rare occurrence

**Comment by goyalyashpal** [comment]
meanwhile, something similar to kate can be done - to pass both the selected files to some external tool like [meld](https://meldmerge.org)

**Comment by Jieiku** [comment]
nothing beats being able to edit as you compare, I just launch the relevant files in Atom when I need to.

**Comment by goyalyashpal** [comment]
> (apparently meld allows you to edit? so maybe that would be neat)

yep, it does :)

**Comment by goyalyashpal** [comment]
- I just today found that even VS Code has an extension for comparing via external comparison tools, and 
- i found it very helpful as i needed the diffs as patches, and the inbuilt diff of VSCode doesn't provide that.

Again, this goes on to reinforce that having this ability to diff with external dedicated tools is still required despite having inbuilt diffs - so, why not have it earlier ;)

> Meld DIff
> Use meld (or other tools like WinMerge, Beyond Compare, ...) to compare files, folders, clipboard or git changes from visual studio code directly
> https://open-vsx.org/vscode/item?itemName=danielroedl.meld-diff

**Comment by LiGoldragon** [comment]
> I already set Lapce as my default git editor.
@josator 

Are you using lapce as a merge tool? I would really like to know how you did it, as that would be enough for me to switch to lapce. Thank you 🙏
